### PR TITLE
feat(compiler): Allow renaming imports with qualified names [LNG-361]

### PR DESCRIPTION
**Pull request by InversionSpaces**
## Description
Refactor internal aqua module management.
Allow users redeclare whole modules, e.g.
```aqua
aqua Lib declares SubLib
use "sublibA" -- brings SubLib.A.* symbols into scope
use "sublibB" -- brings Sublib.B.* symbols into scope
```
declares all `Lib.SubLib.*` symbols.
Allow renaming nested symbols when importing, e.g.
```aqua
aqua Module
-- Brings `Lib.A.Sub.f` into scope as `Lib.B.Another.g`
use A.Sub.f as B.Another.g from "lib"
```
Allow exporting nested imported symbols, e.g.
```aqua
aqua Module
-- exports `Lib.Sub.f` as `g`
export Lib.Sub.f as g
-- brings `Lib.Sub.f` into scope
use "lib"
```

## Proposed Changes
- Refactor `RawContext` handling. Now abilities are "linearized" meaning that ability `A.B.C` is stored in nested abilities `A`, `B` and `C`. Previously it was stored just under key `"A.B.C"`
- Accommodate parser to necessary changes

## Implementation Details
- `SName` (simple name, without dots) and `PName` (path name that may contain dots) are introduced to substitute `String`s and tighten typing

## Checklist
- [x] Corresponding issue has been created and linked in PR title.
- [x] Proposed changes are covered by tests.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [x] I have self-reviewed my code and ensured its quality.
- [x] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: model/raw/src/main/scala/aqua/raw/RawContext.scala [hunk 1]
```diff
@@ -147,10 +207,8 @@ object RawContext {
 
     override def combine(x: RawContext, y: RawContext): RawContext =
       RawContext(
```
Comment (DieMyst) [inline model/raw/src/main/scala/aqua/raw/RawContext.scala:209]: why no exports? maybe add some comment about this

File: model/src/main/scala/aqua/model/AquaContext.scala [hunk 2]
```diff
@@ -121,19 +125,25 @@ case class AquaContext(
...
-      pickOne(name, newName, types, (ctx, el) => ctx.copy(types = el)) |+|
-      pickOne(name, newName, values, (ctx, el) => ctx.copy(values = el)) |+|
-      pickOne(name, newName, abilities, (ctx, el) => ctx.copy(abilities = el)) |+|
-      pickOne(name, newName, services, (ctx, el) => ctx.copy(services = el))
-  }
+  def pick(name: PName, maybeRename: Option[PName]): AquaContext =
+    name.uncons match {
+      case (name, Some(subpath)) => abilities.get(name).map(_.pick(subpath, maybeRename)).orEmpty
+      case (name, None) => {
+        // WARNING: maybeRename here is lost if it is not a simple name
```
Comment (DieMyst) [inline model/src/main/scala/aqua/model/AquaContext.scala:132]: maybe add log for this?

**Comment by DieMyst** [comment]
- will be great to add integration test on feature
- tests that this feature is supported by LSP (or just create task)

File: model/src/main/scala/aqua/model/AquaContext.scala (hunk 2 above)
Comment (InversionSpaces) [reply model/src/main/scala/aqua/model/AquaContext.scala]: Refactored so it is not lost

**Review by DieMyst — APPROVED** [review]

**Pull request merged**